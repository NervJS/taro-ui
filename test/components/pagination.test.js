import Nerv, { findDOMNode } from 'nervjs'
import { renderToString } from 'nerv-server'
import { Simulate, renderIntoDocument } from 'nerv-test-utils'
import AtPagination from '../../.temp/components/pagination/index'

describe('AtPagination Snap', () => {
  it('render AtPagination -- props current', () => {
    const component = renderToString(<AtPagination current={1} />)
    expect(component).toMatchSnapshot()
  })

  it('render AtPagination -- props total', () => {
    const component = renderToString(<AtPagination total={100} />)
    expect(component).toMatchSnapshot()
  })

  it('render AtPagination -- props pageSize', () => {
    const component = renderToString(<AtPagination pageSize={40} />)
    expect(component).toMatchSnapshot()
  })

  it('render AtPagination -- props icon', () => {
    const component = renderToString(<AtPagination icon />)
    expect(component).toMatchSnapshot()
  })
})

describe('AtPagination Event', () => {
  it('AtPagination onPageChange - prev', () => {
    const onPageChange = jest.fn()
    const component = renderIntoDocument(
      <AtPagination total={100} current={2} onPageChange={onPageChange} />
    )
    const doms = findDOMNode(component, 'at-pagination').querySelectorAll(
      '.at-button'
    )
    const prev = doms[0]
    Simulate.click(prev)
    expect(onPageChange).toBeCalled()
    process.nextTick(() => {
      expect(component.state.currentPage).toEqual(1)
    })
  })

  it('AtPagination onPageChange - next', () => {
    const onPageChange = jest.fn()
    const component = renderIntoDocument(
      <AtPagination total={100} current={2} onPageChange={onPageChange} />
    )
    const doms = findDOMNode(component, 'at-pagination').querySelectorAll(
      '.at-button'
    )
    const next = doms[1]
    Simulate.click(next)
    expect(onPageChange).toBeCalled()
    process.nextTick(() => {
      expect(component.state.currentPage).toEqual(3)
    })
  })

  it('AtPagination onPageChange not to be called(disabled prev or next)', () => {
    const onPageChange = jest.fn()
    const component = renderIntoDocument(
      <AtPagination total={20} onPageChange={onPageChange} />
    )
    const doms = findDOMNode(component, 'at-pagination').querySelectorAll(
      '.at-button'
    )
    const prev = doms[0]
    const next = doms[1]
    Simulate.click(prev)
    Simulate.click(next)
    expect(onPageChange).not.toBeCalled()
  })

  it('AtPagination onPageChange params {type, current} - prev', () => {
    const onPageChange = jest.fn()
    const component = renderIntoDocument(
      <AtPagination total={100} current={2} onPageChange={onPageChange} />
    )
    const doms = findDOMNode(component, 'at-pagination').querySelectorAll(
      '.at-button'
    )
    const prev = doms[0]
    Simulate.click(prev)
    expect(onPageChange.mock.calls[0][0].type).toEqual('prev')
    expect(onPageChange.mock.calls[0][0].current).toEqual(1)
  })

  it('AtPagination onPageChange params {type, current} - next', () => {
    const onPageChange = jest.fn()
    const component = renderIntoDocument(
      <AtPagination total={100} onPageChange={onPageChange} />
    )
    const doms = findDOMNode(component, 'at-pagination').querySelectorAll(
      '.at-button'
    )
    const next = doms[1]
    Simulate.click(next)
    expect(onPageChange.mock.calls[0][0].type).toEqual('next')
    expect(onPageChange.mock.calls[0][0].current).toEqual(2)
  })
})
