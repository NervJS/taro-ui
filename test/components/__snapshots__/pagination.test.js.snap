// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AtPagination Snap render AtPagination -- props current 1`] = `"<div class=\\"at-pagination\\"><div class=\\"at-pagination__btn-prev\\">undefined<div class=\\"at-button at-button--small at-button--disabled\\">undefinedundefinedundefinedundefined<div class=\\"at-button__text\\">上一页</div></div></div><div class=\\"at-pagination__number\\"><span class=\\"taro-text at-pagination__number-current\\">1</span>/1</div><div class=\\"at-pagination__btn-next\\">undefined<div class=\\"at-button at-button--small at-button--disabled\\">undefinedundefinedundefinedundefined<div class=\\"at-button__text\\">下一页</div></div></div></div>"`;

exports[`AtPagination Snap render AtPagination -- props icon 1`] = `"<div class=\\"at-pagination at-pagination--icon\\"><div class=\\"at-pagination__btn-prev\\"><div class=\\"at-button at-button--small at-button--disabled\\">undefinedundefinedundefinedundefined<div class=\\"at-button__text\\"><span class=\\"taro-text at-icon at-icon-chevron-left\\"></span></div></div>undefined</div><div class=\\"at-pagination__number\\"><span class=\\"taro-text at-pagination__number-current\\">1</span>/1</div><div class=\\"at-pagination__btn-next\\"><div class=\\"at-button at-button--small at-button--disabled\\">undefinedundefinedundefinedundefined<div class=\\"at-button__text\\"><span class=\\"taro-text at-icon at-icon-chevron-right\\"></span></div></div>undefined</div></div>"`;

exports[`AtPagination Snap render AtPagination -- props pageSize 1`] = `"<div class=\\"at-pagination\\"><div class=\\"at-pagination__btn-prev\\">undefined<div class=\\"at-button at-button--small at-button--disabled\\">undefinedundefinedundefinedundefined<div class=\\"at-button__text\\">上一页</div></div></div><div class=\\"at-pagination__number\\"><span class=\\"taro-text at-pagination__number-current\\">1</span>/1</div><div class=\\"at-pagination__btn-next\\">undefined<div class=\\"at-button at-button--small at-button--disabled\\">undefinedundefinedundefinedundefined<div class=\\"at-button__text\\">下一页</div></div></div></div>"`;

exports[`AtPagination Snap render AtPagination -- props total 1`] = `"<div class=\\"at-pagination\\"><div class=\\"at-pagination__btn-prev\\">undefined<div class=\\"at-button at-button--small at-button--disabled\\">undefinedundefinedundefinedundefined<div class=\\"at-button__text\\">上一页</div></div></div><div class=\\"at-pagination__number\\"><span class=\\"taro-text at-pagination__number-current\\">1</span>/5</div><div class=\\"at-pagination__btn-next\\">undefined<div class=\\"at-button at-button--small\\"><button class=\\"at-button__wxbutton\\" lang=\\"en\\">undefined</button>undefinedundefinedundefined<div class=\\"at-button__text\\">下一页</div></div></div></div>"`;
